x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -243,6 +243,10 @@ void processResult(AutodetectResult result) {
         if (modelPlot != null) {
             bulkResultsPersister.persistModelPlot(modelPlot);
         }
+        Annotation annotation = result.getAnnotation();
+        if (annotation != null) {
+            annotationPersister.persistAnnotation(null, annotation);
```

Comment:
There is no batching here.  I worry that this could be really slow if there is a job with 100000 split field values and we detect periodicity in all 100000 time series at the same time.  To avoid the risk it needs to create bulk requests for these model change annotations, then execute these bulk requests when the next bucket result is seen, when the job is flushed, or when the results stream ends (basically wherever we currently call `bulkResultsPersister.executeRequest()`).

Reply:
> There is no batching here. 

Good point. Let me do it in a separate PR that I'm going to send soon.
Then I'll merge the preparational PR (batching) before this one (model change results).

> Another thing that needs doing as part of this project is to add a subtype enum field to each annotation to record what sort of annotation it is. 

Already sent for review (https://github.com/elastic/elasticsearch/pull/57144).

Reply:
Update:
batching is now implemented in https://github.com/elastic/elasticsearch/pull/57278

Reply:
> then execute these bulk requests when the next bucket result is seen, when the job is flushed, or when the results stream ends

Done.

> (basically wherever we currently call bulkResultsPersister.executeRequest())

There are some calls to `bulkResultsPersister.executeRequest()` which I think do not need accompanying calls to `bulkAnnotationsPersister.executeRequest()`, namely forecasts and quantiles.